src/orders/controller/order.controller.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { OrderService } from '../service/order.service';
+import { OrderMapper } from '../mapper/order.mapper';
+import { Order } from '../domain/order.domain';
+import { OrderDTO } from '../dto/order.dto';
+import { CreateOrderDTO } from '../dto/create-order.dto';
+import { CustomerService } from 'src/customers/service/customer.service';
+import { UpdateOrderDTO } from '../dto/update-order.dto';
+import { ApiResponse } from '@nestjs/swagger';
+import { CustomerMapper } from 'src/customers/mapper/customer.mapper';
+import { OrderDetailMapper } from '../mapper/order-detail.mapper';
+import { LocationService } from 'src/products/service/location.service';
+import { OrderDetail } from '../domain/order-detail.domain';
+import { Location } from 'src/products/domain/location.domain';
+
+@Controller('orders')
+export class OrderController {
+  private readonly orderMapper: OrderMapper;
+  private readonly orderDetailMapper: OrderDetailMapper;
+  private readonly customerMapper: CustomerMapper;
+
+  constructor(
+    private readonly orderService: OrderService,
+    private readonly customerService: CustomerService,
+    private readonly locationService: LocationService,
+  ) {
+    this.orderMapper = new OrderMapper();
+    this.orderDetailMapper = new OrderDetailMapper();
+    this.customerMapper = new CustomerMapper();
+  }
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'The orders were succesfully retrieved',
+  })
+  async getAllOrders(): Promise<OrderDTO[]> {
+    const allOrders: Order[] = await this.orderService.getAllOrders();
+    const allOrdersDTO: OrderDTO[] = [];
+
+    for (const order of allOrders) {
+      const customer = await this.customerService.getCustomerById(
+        order.customer.id,
+      );
+
+      allOrdersDTO.push(
+        this.orderMapper.mapOrderToOrderDTO(
+          order,
+          this.customerMapper.mapCustomerToCustomerDTO(customer),
+        ),
+      );
+    }
+
+    return allOrdersDTO;
+  }
+
+  @Get(':id')
+  @ApiResponse({
+    status: 200,
+    description: 'The order was succesfully retrieved',
+  })
+  @ApiResponse({
+    status: 404,
+    description: 'The order was not found',
+  })
+  async getOrderById(@Param('id') id: string): Promise<OrderDTO> {
+    const order: Order = await this.orderService.getOrderById(id);
+    const customer = await this.customerService.getCustomerById(
+      order.customer.id,
+    );
+
+    return this.orderMapper.mapOrderToOrderDTO(
+      order,
+      this.customerMapper.mapCustomerToCustomerDTO(customer),
+    );
+  }
+
+  @Post()
+  @ApiResponse({
+    status: 201,
```

Comment:
Probably here a 404 is missing. I see you have the check in the service for the customer and you throw the error.